esmvalcore/experimental/config/_config_validators.py

```diff
@@ -111,6 +111,59 @@ def func(inp):
     return func
 
 
+@lru_cache()
+def _tuplify_validator(scalar_validator,
+                       allow_stringlist=False,
+                       *,
+                       n_items=None,
+                       docstring=None):
+    """Apply the validator to a list."""
+    def func(inp):
+        if isinstance(inp, str):
+            try:
+                inp = tuple(
+                    scalar_validator(val.strip()) for val in inp.split(',')
+                    if val.strip())
+            except Exception:
+                if allow_stringlist:
+                    # Sometimes, a list of colors might be a single string
+                    # of single-letter colornames. So give that a shot.
+                    inp = tuple(
+                        scalar_validator(val.strip()) for val in inp
+                        if val.strip())
+                else:
+                    raise
+        # Allow any ordered sequence type -- generators, np.ndarray, pd.Series
+        # -- but not sets, whose iteration order is non-deterministic.
+        elif isinstance(inp,
+                        Iterable) and not isinstance(inp, (set, frozenset)):
+            # The condition on this tuple comprehension will preserve the
+            # behavior of filtering out any empty strings (behavior was
+            # from the original validate_stringlist()), while allowing
+            # any non-string/text scalar values such as numbers and arrays.
+            inp = tuple(
+                scalar_validator(val) for val in inp
+                if not isinstance(val, str) or val)
+        else:
+            raise ValidationError(
+                f"Expected str or other non-set iterable, but got {inp}")
+        if n_items is not None and len(inp) != n_items:
+            raise ValidationError(f"Expected {n_items} values, "
+                                  f"but there are {len(inp)} values in {inp}")
+        return inp
+
+    try:
+        func.__name__ = "{}tuple".format(scalar_validator.__name__)
+    except AttributeError:  # class instance.
+        func.__name__ = "{}Tuple".format(type(scalar_validator).__name__)
+    func.__qualname__ = func.__qualname__.rsplit(".",
+                                                 1)[0] + "." + func.__name__
+    if docstring is not None:
+        docstring = scalar_validator.__doc__
+    func.__doc__ = docstring
+    return func
```

Comment:
I don't think it is necessary to duplicate the entire `_listify_validator` code here, how about this:

```suggestion
def _tuplify_validator(scalar_validator, 
                       allow_stringlist=False,
                       *,
                       n_items=None,
                       docstring=None):
    """Apply validator to a list and return a tuple."""
     _list_validator = _listify_validator(scalar_validator, allow_stringlist, n_items=n_items, docstring=docstring)
     def func(inp):
         lst = _list_validator(inp)
         return tuple(lst)
     return func
```

Reply:
I had another approach using an `as_tuple` toggle on the `_listify_validator`, and then adding a `if as_tuple: return tuple(inp)` statement at the end. I think I prefer the version above.

Reply:
Hm, I see your point, but would in principle rather like to avoid constructing unneeded objects. How about an option as you propose, but treating it as a selector for the iterable type used in the generator expression? I.e.
```
iterablizer = tuple if as_tuple else list
...
inp = iterablizer(...)
```

Reply:
Though I also recognize that it will hardly make any real difference in the config context and that this issue is rather tangential here and must not delay this PR.

Reply:
That also works for me 👍 The thing I want to avoid is that we duplicate the entire `_listify_validator`.